vm/devices/storage/storvsc_driver/src/lib.rs

```diff
@@ -0,0 +1,920 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Storvsc driver for use as a disk backend.
+
+#[cfg(feature = "test")]
+pub mod test_helpers;
+
+#[cfg(not(feature = "test"))]
+mod test_helpers;
+
+use futures::FutureExt;
+use futures_concurrency::future::Race;
+use guestmem::AccessError;
+use guestmem::MemoryRead;
+use guestmem::ranges::PagedRange;
+use mesh_channel::Receiver;
+use mesh_channel::RecvError;
+use mesh_channel::Sender;
+use slab::Slab;
+use task_control::AsyncRun;
+use task_control::InspectTask;
+use task_control::StopTask;
+use task_control::TaskControl;
+use thiserror::Error;
+use tracing_helpers::ErrorValueExt;
+use vmbus_async::queue;
+use vmbus_async::queue::CompletionPacket;
+use vmbus_async::queue::DataPacket;
+use vmbus_async::queue::ExternalDataError;
+use vmbus_async::queue::IncomingPacket;
+use vmbus_async::queue::OutgoingPacket;
+use vmbus_async::queue::PacketRef;
+use vmbus_async::queue::Queue;
+use vmbus_channel::RawAsyncChannel;
+use vmbus_ring::OutgoingPacketType;
+use vmbus_ring::PAGE_SIZE;
+use vmbus_ring::RingMem;
+use vmcore::vm_task::VmTaskDriverSource;
+use zerocopy::FromBytes;
+use zerocopy::Immutable;
+use zerocopy::IntoBytes;
+use zerocopy::KnownLayout;
+
+/// Storvsc to provide a backend for SCSI devices over VMBus.
+pub struct StorvscDriver<T: Send + Sync + RingMem> {
+    storvsc: TaskControl<StorvscState, Storvsc<T>>,
```

Comment:
If I'm reading `TaskControl` correctly, the first type is the *Task* and the second type is the *State*, but that seems backwards here.